views/pages/AdminPage.jsx

```diff
@@ -1,696 +1,201 @@
-          <Dialog
-            modal={false}
-            open={this.state.openReset}
-            onClose={this.closeReset}
-          >
-            <DialogTitle
-              style={{
-                backgroundColor: 'rgba(0,0,0,0.7)',
-              }}
-            >
-              <Typography variant="headline" style={{ color: 'white' }}>
-                Reset user password
-              </Typography>
-            </DialogTitle>
-            <DialogContent>
-              <Subheader
-                style={{
-                  padding: '0px',
-                }}
-              >
-                {this.state.users[this.state.resetUser]
-                  ? 'Reset key for user: ' +
-                    this.state.users[this.state.resetUser]['uid']
-                  : ''}
-              </Subheader>
-              <p>
-                {this.state.users[this.state.resetUser]['reset_key']
-                  ? this.state.users[this.state.resetUser]['reset_key']
-                  : ''}
-              </p>
-            </DialogContent>
-          </Dialog>
-        ) : null}
-      </>
-    )
-  }
+  const updateUserInForm = (rowIndex, updatedUser) =>
+    setValue(`users.${rowIndex}`, updatedUser)
+
+  useEffect(() => {
+    handleResize()
+    fetchStudies()
+
+    window.addEventListener('resize', handleResize)
+
+    return () => window.removeEventListener('resize', handleResize)
+  }, [])
+
+  useEffect(() => {
+    const searchOptionsFromUsers = UsersModel.createUserFriendList(users)
+    const createFormValuesFromUsers =
+      users.map((user) => {
+        return {
+          ...user,
+          role: { value: user.role, label: user.role },
+          access: UsersModel.createUsersAccesList(user.access),
+        }
+      }) || []
+
+    setSearchOptions(searchOptionsFromUsers)
+    setValue(usersKey, createFormValuesFromUsers)
+  }, [users])
+
+  return (
+    <>
+      <NoSsr>
+        <Select
+          autoFocus={true}
+          classes={classes}
+          components={components}
+          isMulti
+          onChange={(e) => setValueofOptions(e)}
+          options={searchOptions}
+          placeholder="Search users"
+          value={seearchOptionsValue}
+        />
+      </NoSsr>
+      <AdminForm
+        classes={classes}
+        control={control}
+        currentRowIndex={currentRowIndex}
+        height={dimensions.height}
+        onClose={handleClose}
+        options={studies}
+        onAccess={handleUserRole}
+        onDeleteUser={handleUserRemove}
+        onFilter={handleUserListFilter}
+        onResetPassword={handleResetPassword}
+        onSubmit={handleSubmit(handleUserUpdate)}
+        onUpdateUserAccess={handleUpdateUserAccess}
+        openAccessFields={openAccessModal}
+        openResetKeyFields={openResetKeyModal}
+        resetKey={resetKeyValue}
+        rowCount={
+          seearchOptionsValue.length
+            ? seearchOptionsValue.length
+            : fields.length
+        }
```

Comment:
can you explain this logic? It's not obvious to me why we need to either use `seearchOptionsValue` or `fields`.